wsagent/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitUrlUtils.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.git;
+
+import java.util.regex.Pattern;
+
+/**
+ * Utility class for working with Git urls.
+ *
+ * @author Vladyslav Zhukovskii
+ * @author Kevin Pollet
+ */
+public class GitUrlUtils {
+    public static final Pattern GIT_SSH_URL_PATTERN =
+            Pattern.compile("((((git|ssh)://)(([^\\\\/@:]+@)??)[^\\\\/@:]+)|([^\\\\/@:]+@[^\\\\/@:]+))(:|/)[^\\\\@:]+");
```

Comment:
I have only moved this class to another package, but I added a test for it: https://github.com/eclipse/che/blob/bddc03c290c1801854990f12cb07b0d889d7913d/wsagent/che-core-api-git/src/test/java/org/eclipse/che/api/git/GitUrlUtilsTest.java